### ISSUE TITLE
Route propagation

### ISSUE THREAD

**Issue by igorvpcleao** [cloudnativelabs/kube-router#222]
Hi everyone,

I have a Kubernetes app which just do nslookups to popular websites, such as google, bing, github, etc.
When no node is available to schedule it, cluster autoscaler launches a new Ec2 instance and I have to wait till it gets ready.
The problem is, once the new node gets ready and my pod is allocated to this node, sometimes it takes a few minutes to have routes propagated to it. By looking at my pod logs I can see nslookups are failing. Then, here it is the output when I run `route` on this new node:
```
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         ip-10-0-62-1.ec 0.0.0.0         UG    0      0        0 eth0
10.0.62.0       0.0.0.0         255.255.255.0   U     0      0        0 eth0
100.96.54.0     0.0.0.0         255.255.255.0   U     0      0        0 kube-bridge
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
```
As soon as my pod stops failing nslookups, I check `route` again and it seems much more complex:
```
ernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         ip-10-0-62-1.ec 0.0.0.0         UG    0      0        0 eth0
10.0.62.0       0.0.0.0         255.255.255.0   U     0      0        0 eth0
100.96.0.0      ip-10-0-62-100. 255.255.255.0   UG    0      0        0 eth0
100.96.1.0      ip-10-0-62-101. 255.255.255.0   UG    0      0        0 eth0
100.96.2.0      ip-10-0-62-102.e 255.255.255.0   UG    0      0        0 eth0
100.96.5.0      ip-10-0-62-103. 255.255.255.0   UG    0      0        0 eth0
[...]
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
```
Does it seem to be a kube-router issue? Do you know how I can mitigate this?

Thanks in advance!

**Comment by murali-reddy** [comment]
Seems like a kube-router issue. There is issue opened #202. Basically there is time delay for the routes to propaged to new node. Likely connectivity is not establised between your pod and kube-dsn.

I am trying to work on 202, hopefully that should fix this issue.

**Comment by igorvpcleao** [comment]
Thx, @murali-reddy! Could it be a limitation due to the `full node-to-node mesh`? Do already know what needs to be modified in order to increase performance? Let me know if I can contribute to it somehow.

**Comment by murali-reddy** [comment]
On top of my head if you set `--routes-sync-period` to more aggressive value (currently its 60s) to smaller value convergence should happen fast. Please try if that works. There are may be other GoBGP knobs that are not fully utlized to fasten things up. I will have to test a bit to understand where (From kube-router or GoBGP) the root cause is.

**Comment by igorvpcleao** [comment]
Thnaks @murali-reddy! I tried it last week, but didn't have time to go deeper and double-check myself. Just tried it again reducing `--routes-sync-period` to `10s` and rolling restarting kube-router in every node. After that I scaled out a certain deploy and the problem seems to persists.

**Comment by murali-reddy** [comment]
@igorvpcleao i will try to reproduce the issue.

**Comment by wu0407** [comment]
the situcation with me:  kube-route first propagation  route to  external route  cost five minutes after kube-route  started,  set --routes-sync-period not useful.
this is capture the traffic  kube-route as rr 192.168.10.11  with  192.168.254.66(external route)

[kube-route-201907031958.zip](https://github.com/cloudnativelabs/kube-router/files/3354984/kube-route-201907031958.zip)

**Comment by b-abadie** [comment]
@wu0407 We are facing the exact same problem : when the BGP session is re-established, kube-router waits 6 minutes after session is established (packet # 16) to send its routes (packet # 75). And in our setup, gobgp showed the routes as being advertised (`gobgp neighbor X.X.X.X`), while a packet dump showed no UPDATE message before the 6 minutes mark (just like yours).

We observed that this behavior is *not present* when we disable BGP graceful restart (i.e. no `--bgp-graceful-restart`).

However, I am not sure this is the same issue as the one described in the comments above : we have an issue with the routes getting propagated to the external router whereas they seem to have an issue with the routes being learnt by the node.

**Comment by wu0407** [comment]
the relate pull request https://github.com/cloudnativelabs/kube-router/pull/753

**Comment by aauren** [comment]
This should be resolved with #753 

Please test a new version of kube-router and re-open if necessary.
